elastic/kibana
# [Infra UI] Node Details using Asset Details component

part of [#162898](https://github.com/elastic/kibana/issues/162898)

## Summary

This PR implements the new node details page for hosts. It uses the Asset Details component, which is the same used in the Hosts View flyout. 

https://github.com/elastic/kibana/assets/2767137/0d3d2b2d-ed82-4275-b424-ddae82cdbddc

It **does not** change the behaviour of other asset types (kubernetes, container, aws...). The main change this PR introduces is internalizing the Data Views logic so that clients don't have to worry about passing it to the Asset Details.

The fundamental change between flyout and page is the template. Everything else is shared. 

Functional tests will be added in a separate PR.

### How to test this PR
- Setup a local Kibana instance
- Navigate to `Infrastructure` > `Hosts`
  - Click on a host name and check if the new Node Details page opened
  - Return to the hosts page and open a flyout
  - Click on `Open as page` and check if the new Node Details page opened
- Navigate to `Infrastructure` > `Inventory`
  - With `Show: Host` selected, click on a waffle item to open the flyout, click on  `Open as page` and check if the new Node Details page opened
  - Select another asset type in `Show` , open the flyout and check if the old version of the Node Details Page will show

#### Storybook
`yarn storybook infra`


### For reviewers

Errors in the new page will be handled differently from what the current Node Details page. i.e: accessing the page with invalid asset id, won't show this page

<img width="1443" alt="image" src="https://github.com/elastic/kibana/assets/2767137/93de3caf-ac63-4a34-a90d-226b212096fe">
